restx-samplest/src/main/java/samplest/autostartable/AutoStartableTestComponent.java

```diff
@@ -14,9 +14,9 @@
 */
 @Component
 public class AutoStartableTestComponent implements AutoStartable, AutoCloseable {
-    private static int closed;
-    private static int started;
-    private static int instanciated;
+    private int closed;
+    private int started;
+    private int instanciated;
```

Comment:
I still think we shouldn't change that. We should better change the way the test is called in `AutoStartableTest`, maybe using a new factory there would be enough.

Reply:
Hi,

I first tried to use a new `Factory` but I couldn't make it work. [Here is a change with a new Factory](https://github.com/nhumblot/restx/commit/1865bc87aca19e1c8765c8997042fb3b6ac76d51).

When launching a `mvn verify`, the test is failing.
```
Failed tests: 
  AutoStartableTest.should_handle_auto_startable_in_dev_mode:33 expected:<...called: 1 started: 1[ closed: 0 instanciated: ]1 serverId: Jetty8#1...> but was:<...called: 1 started: 1[11 closed: 110 instanciated: 11]1 serverId: Jetty8#1...>
Tests run: 83, Failures: 1, Errors: 0, Skipped: 0
```
Test log:
```
should_handle_auto_startable_in_dev_mode(samplest.autostartable.AutoStartableTest)  Time elapsed: 6.885 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<...called: 1 started: 1[ closed: 0 instanciated: ]1 serverId: Jetty8#1...> but was:<...called: 1 started: 1[11 closed: 110 instanciated: 11]1 serverId: Jetty8#1...>
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at samplest.autostartable.AutoStartableTest.should_handle_auto_startable_in_dev_mode(AutoStartableTest.java:33)
```


The test is not getting the expected count of `started`, `closed` and `instantiated` values.

One thing to note, when this test is executed alone, it passes.

I think it fails in `mvn verify` because we execute multiple tests. In this situation, `AutoStartableTest` is not the first test to run. So we start a server multiple times but JUnit keep the same class loader so static values are kept between tests.

So it means if we need to keep these fields static, we should also execute each unit test with a separate class loader ? This way, static fields will be reset and we can enforce the value ?

Or another solution, seems simpler and better to me, would be, at the start of the test, we get the `started`, `closed` and `instantiated` values, store them as local variables during the test and we use them as base values instead of supposing these values are all at 0 before we start the test.

After your message, I understand your concern, and I think I see the issue you are pointing. Is it because the dev mode throw and create components after each request ?

Please tell me if one of the two solutions exposed satisfy you and which one has your preference. I can work on that and update this pull request accordingly.

Reply:
You're right, thanks for this investigation.

I think the simplest solution would be to store the values before the test and compare with them, or add a `reset` method on AutoStartable which sets everything to 0, and call it in the test. Neither of these solutions would make the test able to run in parallel but at least it should solve the problem and still perform the intent of the test.

Reply:
This test has been fixed accordingly to our discussion. Change in 69ec801f0dc3dede6a324a95f62c13c4dc8a077f. You can add comments on this change if needed.

I updated to the current version of `master`, rebased this branch on it, and dropped the previous commit on `AutoStartableTest` fix that generated this comment.

I will start working on MongoDB startup onward.

Reply:
Perfect!